flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,434 @@
+                        db_session.expires >= datetime.utcnow()):
+                    # Found a valid session in the DB - use it
+                    self._session.cookies.update(db_session.cookies)
+                    self.authkey = db_session.authkey
+                    self.passkey = db_session.passkey
+                    log.info("Logged in to %s using cached session", self.base_url)
+                    return
+
+        # Forcing a re-login or no session in DB - log in using provided creds
+        url = "{}/login.php".format(self.base_url)
+        data = {
+            'username': self.username,
+            'password': self.password,
+            'keeplogged': 1,
+        }
+        r = self._session.post(url, data=data, allow_redirects=False)
+        if not r.is_redirect or r.next.url != "{}/index.php".format(self.base_url):
+            msg = "Failed to log into {}".format(self.base_url)
+            for x in DETECT_2FA:
+                if x in r.text:
+                    msg += " - Accounts using 2FA are currently not supported"
+                    break
+            raise PluginError(msg)
+
+        accountinfo = self.request(no_login=True, action='index')
+        self.authkey = accountinfo['authkey']
+        self.passkey = accountinfo['passkey']
+        log.info("Logged in to %s", self.base_url)
+
+        # Store new session in DB
+        log.debug("Storing session info in DB")
+        with FlexgetSession() as session:
+            expires = None
+            for c in self._session.cookies:
+                if c.name == "session":
+                    expires = datetime.utcfromtimestamp(c.expires)
+            db_session = GazelleSession(username=self.username,
+                                        base_url=self.base_url,
+                                        cookies=dict(self._session.cookies),
+                                        expires=expires, authkey=self.authkey,
+                                        passkey=self.passkey)
+            session.merge(db_session)
+
+    def request(self, no_login=False, **params):
+        """
+        Make an AJAX request to the API
+
+        If `no_login` is True, logging in will not be attempted if the request
+        is redirected to the login page.
+
+        Adapted from https://github.com/isaaczafuta/whatapi
+        """
+        if 'action' not in params:
+            raise ValueError("An 'action' is required when making a request")
+
+        ajaxpage = "{}/ajax.php".format(self.base_url)
+        r = self._session.get(ajaxpage, params=params, allow_redirects=False)
+        if (not no_login and r.is_redirect and
+                r.next.url == "{}/login.php".format(self.base_url)):
+            log.warning("Redirected to login page, reauthenticating and trying again")
+            self.authenticate(force=True)
+            return self.request(no_login=True, **params)
+
+        if r.status_code != 200:
```

Comment:
should be moved to exception handling

Reply:
This response returns a 200 in a success case. Any other 3xx or 2xx response, while not considered an error to `requests`, is considered an error at this point in execution.